catroid/src/main/java/org/catrobat/catroid/devices/arduino/ArduinoImpl.java

```diff
@@ -222,30 +222,36 @@ public void setAnalogArduinoPin(int pin, int value) {
 		sendAnalogFirmataMessage(pin, value);
 	}
 
+	public static int setBit(int number, int index, int value) {
+		if (value == 0) {
+			return number & ~(1 << index);
+		} else {
+			return number | (1 << index);
+		}
+	}
+
```

Comment:
Please create a test for this method.

There is one ui test class for the Arduino feature. I've created a ticket because the Arduino feature can and should be tested by unit tests. [CAT-2196](https://jira.catrob.at/browse/CAT-2196).

@jobermue @thmq, is there someone who can review the Arduino changes? Someone who is familiar with the Arduino code and also can test it with an Arduino.